src/gt4py/next/ffront/decorator.py

```diff
@@ -171,7 +171,9 @@ class Program:
     past_node: past.Program
     closure_vars: dict[str, Any]
     definition: Optional[types.FunctionType] = None
-    backend: Optional[ppi.ProgramExecutor] = None
+    backend: None | eve_utils.NOTHING | ppi.ProgramExecutor = (
```

Comment:
TODO: spread this change to everywhere where we use the default backend or find another strategy